### DISCUSSION TITLE
About booting and self starting issues under Windows

### DISCUSSION THREAD

**Discussion by EHW-liao** [tauri-apps/tauri#7051] [Q&A]
May I ask how to achieve the startup and self start of the tauri application, and only display the system tray without displaying the main application window.

Edit by Maintainer:
Content of [duplicate question](https://github.com/tauri-apps/tauri/discussions/7050) by the same user:
> Writing the path of the tauri application in the registry path can achieve booting, but I only want to display it as the system tray instead of displaying the application homepage at the same time. How can I achieve this? Can parameters be passed in through the registry to control not displaying the interface when the software starts?

**Comment by FabianLars** [comment]
We have a plugin for the autostart part of the question: https://github.com/tauri-apps/tauri-plugin-autostart/

Not spawning a window just comes down to not defining one in tauri.conf.json, but you'll have to explicitly keep the event loop running in rust so that the app doesn't exit: https://github.com/tauri-apps/tauri/blob/b02fc90f450ff9e9d8a35ee55dc1beced4957869/examples/api/src-tauri/src/main.rs#L240-L244

> Can parameters be passed in through the registry

The plugin does not support this but if you already set the regkey yourself yes, you can. Small example (first google result): https://stackoverflow.com/a/19299564 - you should then be able to read that via https://doc.rust-lang.org/std/env/fn.args.html
